Sources/SwiftSyntaxBuilder/DeclSyntaxParseable.swift

```diff
@@ -0,0 +1,57 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2023 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import SwiftSyntax
+
+/// Adds an initializer that allows the creation of declaration from string
+/// interpolations.
+///
+/// - Warning: This protocol is considered an implementation detail. Do not rely
+///   on it, only the initializer that it adds.
+public protocol DeclSyntaxParseable: DeclSyntaxProtocol {}
+public extension DeclSyntaxParseable {
```

Comment:
Because, for example, `AccessorDeclSyntax` does not go through `parseDecl`, it has its own parser entry function. And thus, trying to call `parser.parseDecl` and casting it to `AccessorDeclSyntax` wouldn’t work.